pkg/operator/staticpod/controller/guard/guard_controller.go

```diff
@@ -0,0 +1,254 @@
+		for _, pod := range pods {
+			operands[pod.Spec.NodeName] = pod
+		}
+
+		for _, node := range nodes {
+			if _, exists := operands[node.Name]; !exists {
+				klog.Errorf("Missing operand on node %v", node.Name)
+				errs = append(errs, fmt.Errorf("Missing operand on node %v", node.Name))
+				continue
+			}
+
+			if operands[node.Name].Status.PodIP == "" {
+				klog.Errorf("Missing PodIP in operand %v on node %v", operands[node.Name].Name, node.Name)
+				errs = append(errs, fmt.Errorf("Missing PodIP in operand %v on node %v", operands[node.Name].Name, node.Name))
+				continue
+			}
+
+			klog.V(5).Infof("Rendering guard pod for operand %v on node %v", operands[node.Name].Name, node.Name)
+
+			pod := resourceread.ReadPodV1OrDie(bindata.MustAsset(filepath.Join("pkg/operator/staticpod/controller/guard", "manifests/guard-pod.yaml")))
+
+			pod.ObjectMeta.Name = getGuardPodName(c.podResourcePrefix, node.Name)
+			pod.ObjectMeta.Namespace = c.targetNamespace
+			pod.Spec.NodeName = node.Name
+			pod.Spec.Containers[0].Image = c.installerPodImageFn()
+			pod.Spec.Containers[0].ReadinessProbe.HTTPGet.Host = operands[node.Name].Status.PodIP
+			// The readyz port as string type is expected to be convertible into int!!!
+			readyzPort, err := strconv.Atoi(c.readyzPort)
+			if err != nil {
+				panic(err)
+			}
+			pod.Spec.Containers[0].ReadinessProbe.HTTPGet.Port = intstr.FromInt(readyzPort)
+
+			actual, err := c.podGetter.Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+			if err == nil {
+				// Delete the pod so it can be re-created. ApplyPod only updates the metadata part of the manifests, ignores the rest
+				delete := false
+				if actual.Spec.Containers[0].Image != pod.Spec.Containers[0].Image {
+					klog.V(5).Infof("Guard Image changed, deleting %v so the guard can be re-created", pod.Name)
+					delete = true
+				}
+				if actual.Spec.Containers[0].ReadinessProbe.HTTPGet.Host != pod.Spec.Containers[0].ReadinessProbe.HTTPGet.Host {
+					klog.V(5).Infof("Operand PodIP changed, deleting %v so the guard can be re-created", pod.Name)
+					delete = true
+				}
+				if delete {
+					_, _, err = resourceapply.DeletePod(ctx, c.podGetter, syncCtx.Recorder(), pod)
+					if err != nil {
+						klog.Errorf("Unable to delete Pod for immidiate re-creation: %v", err)
+						errs = append(errs, fmt.Errorf("Unable to delete Pod for immidiate re-creation: %v", err))
+						continue
+					}
+				}
+			} else if !apierrors.IsNotFound(err) {
+				errs = append(errs, err)
+				continue
+			}
+
+			_, _, err = resourceapply.ApplyPod(ctx, c.podGetter, syncCtx.Recorder(), pod)
+			if err != nil {
+				klog.Errorf("Unable to apply pod %v changes: %v", pod.Name, err)
+				errs = append(errs, fmt.Errorf("Unable to apply pod %v changes: %v", pod.Name, err))
+			}
+		}
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+func WithIsSNOCheck(infraLister configv1listers.InfrastructureLister, neg bool) resourceapply.ConditionalFunction {
```

Comment:
Do we really need `neg` parameter to negate the result, it's not that hard to add `!` before the invocation, is it?